docs/src/osdk/reference/manifest.md

```diff
@@ -35,61 +35,139 @@ path fields will be relative to the manifest's enclosing directory.
-boot                        # Overriding [boot]
-grub                        # Overriding [grub]
-qemu                        # Overriding [qemu]
-[test]                      # Special settings for testing, which will override default ones
-build                       # Overriding [build]
-boot                        # Overriding [boot]
-grub                        # Overriding [grub]
-qemu                        # Overriding [qemu]
+supported_archs = ["x86_64", "riscv64"]     # <2>
+
+# The common options for all build, run and test subcommands 
+[build]                                     # <3>
+features = ["no_std", "alloc"]              # <4>
+profile = "dev"                             # <5>
+strip_elf = false                           # <6>
+[boot]                                      # <7>
+method = "qemu-direct"                      # <8>
+kcmd_args = ["SHELL=/bin/sh", "HOME=/"]     # <9>
+init_args = ["sh", "-l"]                    # <10>
+initramfs = "path/to/it"                    # <11>
+[grub]                                      # <12>  
+mkrescue_path = "path/to/it"                # <13>
+protocol = "multiboot2"                     # <14> 
+display_grub_menu = false                   # <15>
+[qemu]                                      # <16>
+path = "path/to/it"                         # <17>
+args = "-machine q35 -m 2G"                 # <18>
+
+# Special options for run subcommand
+[run]                                       # <19>
+[run.build]                                 # <3>
+[run.boot]                                  # <7>
+[run.grub]                                  # <12>
+[run.qemu]                                  # <16>
+
+# Special options for test subcommand
+[test]                                      # <20>
+[test.build]                                # <3>
+[test.boot]                                 # <7>
+[test.grub]                                 # <12>
+[test.qemu]                                 # <16>
 # ----------------------- end of the default schema settings ----------------------------
 
-[schema."user_custom_schema"]
-#...                        # All the other fields in the default schema. Missing but
-                            # needed values will be firstly filled with the default
-                            # value then the corresponding field in the default schema
+# A customized schema settings
+[schema."custom"]                           # <21>
+[schema."custom".build]                     # <3>
+[schema."custom".run]                       # <19>
+[schema."custom".test]                      # <20>
 ```
 
 Here are some additional notes for the fields:
 
-1. The arguments provided will be passed to the guest kernel.
+1. The type of current crate.
+
+    Optional. The default value is `kernel`. 
+    
+    Possible values are `library` or `kernel`.
+
+2. The architectures that can be supported.
+
+    Optional. By default, the architecture of the host machine will be used.
```

Comment:
By default OSDK will regard all the settings in this scheme (or the default scheme) to be architectural-independent. That is, supporting all architectures.

When building or running, if not specified in the CLI, the architecture of the host machine will be used.